### PR TITLE
Refresh website

### LINKED ISSUE #2194: [Improvement] Document Git repository requirements
Seeing how getting GitVersion to work in different CI servers (Jenkins seems particularly problematic) is a recurring problem, it would be great if we documented GitVersion’s requirements for successfully generating a version number such as:

1. An unshallow clone.
2. An existing local `master` branch.
3. Possibly an existing local `develop` branch (for GitFlow).
4. Should be [switched](https://git-scm.com/docs/git-switch) to a named, existing branch pointing to the commit being built (i.e. no detached `HEAD`).
5. `Git_Branch` as an available environment variable.
6. If using a `GitVersion.yml` file, that file should be checked out otherwise it won't be found by GitVersion and default config will apply.

Possibly related issues: #1335, #2180, #2118, #2115, #1857, #2233.

### LINKED ISSUE #2326: [Improvement] Detailing package alternatives
Add a page to the documentation that contains up to date information regarding all the different package alternatives GitVersion offers, where they can be found and why one should choose which.

### NuGet

* MSBuild Task: [GitVersionTask](https://www.nuget.org/packages/GitVersionTask/)
* `dotnet tool`: [GitVersionTool](https://www.nuget.org/packages/GitVersion.Tool/)
* Standalone `.exe`: [GitVersion.CommandLine](https://www.nuget.org/packages/GitVersion.CommandLine/)

### Azure DevOps

* [GitTools](https://marketplace.visualstudio.com/items?itemName=gittools.gittools)

### GitHub Actions

* [GitTools actions](https://github.com/marketplace/actions/gittools)

### Docker

* [gittools/gitversion](https://hub.docker.com/r/gittools/gitversion)

### Homebrew

* [gitversion](https://formulae.brew.sh/formula-linux/gitversion)

### Chocolatey

* [GitVersion.Portable](https://chocolatey.org/packages/GitVersion.Portable)

### LINKED ISSUE #2334: [Improvement] Document that ignore only affects the version source
As written in https://github.com/GitTools/GitVersion/issues/2122#issuecomment-642900931 we should document that `IgnoreConfig` only affects GitVersion's search for a version source, not how the ignored commit(s) are counted towards the final verison number.

### LINKED ISSUE #2365: [Docs] How to install on ubuntu. Why so little documentation on Linux installation?
Apologies in advance. I will admit I am a noob with both gitversion and ubuntu. So maybe I am missing something. 

I am trying to install gitversion command line in Ubuntu 18.04.4 LTS. The documentation [here](https://gitversion.net/docs/usage/command-line) says we need to install` mono-complete` and run  `mono gitversion.exe`. 

How do I get the exe file? downloading and uncompressing the tar.gz file from [here ](https://github.com/GitTools/GitVersion/releases) gave me just a single file named `gitversion`, I do not see any exe file. 

```
wget https://github.com/GitTools/GitVersion/releases/download/5.3.7/gitversion-ubuntu.18.04-x64-5.3.7.tar.gz
tar -xvf gitversion-ubuntu.18.04-x64-5.3.7.tar.gz
```

A little more guideline in docs would be helpful.  

Thanks in advance

### LINKED ISSUE #2503: [Docs] Update the docs to reflect the current state of the code base

### LINKED ISSUE #2551: GitVersion should rely on existing botnet SDKs rather than downloading it using brew
I recognized that when I install GitVersion using brew on macos, it downloads a separate dotnet SDK using brew as well. Instead, it should rely on the existing one. 

## Detailed Description
I faced a problem that after installing GitVersion using brew, my Visual Studio Code Omnisharp could not find the installed SDKs anymore. 

After doing some research, I deleted dotnet using brew and reinstalled dotnet using the official installer. After that my Visuals Studio Code Omnisharp found the SDK again, but my GitVersion was not working anymore. 

I assume that installing GitVersion using brew, not only installs dotnet via brew, but also changes Environment that is responsible for pointing to a specific dotnet SDK version when invoking the dotnet command. Probable it changes the PATH in a way that the dotnet version installed via brew gets found first, before the one installed via the official installer. 

## Context
The importance of this is the following. With the described scenario I had to choose if I want to have GitVersion working or my Visual Studio Code Omnisharp. Both were not working at the same time. I had to create a symlink in order to tell GitVersion to use the SDK installed via the official installer.  

```shell
sudo mkdir -p /usr/local/opt/dotnet/bin
sudo ln -s /usr/local/share/dotnet/dotnet /usr/local/opt/dotnet/bin/dotnet
```

## Possible Implementation
* Installing dotnet as well could be made optional using a flag
* The SDK could be specified using an Env variable 
* GitVersion could first check if there is any dotnet SDK already installed before wanting to install a fresh one

### PR DESCRIPTION
**Pull request by asbjornu**
## Description

This PR refreshes the website to match the logo created by @LazaroOnline in #2508. It also updates the documentation so it resolves #2326.

## Related Issue

- Builds on #2508.
- Resolves #2326.
- Resolves #2194.
- Resolves #2503.
- Resolves #2334.
- Resolves #2551.
- Resolves #2365.

## Motivation and Context

The documentation was chaotic and outdated. This PR does not complete that work, but it improves it a whole lot.

## How Has This Been Tested?

- I've made the arguments documentation shared between command line and the docs.
- I've added [HTMLProofer](chabad360/htmlproofer@master) tests that verify links in the generated documentation.

## Screenshots (if appropriate):

front page

usage

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by asbjornu** [comment]
I'm having some [Cake problems](https://github.com/GitTools/GitVersion/pull/2599/checks?check_run_id=1944253116#step:4:110):

```
Cake Setup
   at Cake.Core.Tooling.Tool`1.Run(TSettings settings, ProcessArgumentBuilder arguments, ProcessSettings processSettings, Action`1 postAction) in C:\projects\cake\src\Cake.Core\Tooling\Tool.cs:line 103
   at Cake.Common.Tools.GitVersion.GitVersionRunner.Run(GitVersionSettings settings) in C:\projects\cake\src\Cake.Common\Tools\GitVersion\GitVersionRunner.cs:line 59
   at Cake.Common.Tools.GitVersion.GitVersionAliases.GitVersion(ICakeContext context, GitVersionSettings settings) in C:\projects\cake\src\Cake.Common\Tools\GitVersion\GitVersionAliases.cs:line 121
   at Submission#0.GitVersion(GitVersionSettings settings)
   at Submission#0.GetVersion(BuildParameters parameters)
   at Submission#0.<>c__DisplayClass0_6.<<Initialize>>b__114()
	Message:	GitVersion: Process returned an error (exit code 1).
	Data:	[  ]
	InnerException:	
	HelpLink:	
	Source:	Cake.Core
	HResult:	-2146233088
```

And:

```
Teardown error: System.NullReferenceException: Object reference not set to an instance of an object.
   at Cake.Core.CakeDataService.Get[TData]() in C:\projects\cake\src\Cake.Core\CakeDataService.cs:line 36
   at Cake.Core.CakeEngineActions.<>c__DisplayClass21_0`1.<RegisterTeardown>b__0(ITeardownContext context) in C:\projects\cake\src\Cake.Core\CakeEngineActions.cs:line 64
   at Cake.Core.DefaultExecutionStrategy.PerformTeardown(Action`1 action, ITeardownContext teardownContext) in C:\projects\cake\src\Cake.Core\DefaultExecutionStrategy.cs:line 57
   at Cake.Core.CakeEngine.PerformTeardown(IExecutionStrategy strategy, ICakeContext context, Stopwatch stopWatch, CakeReport report, Boolean exceptionWasThrown, Exception thrownException) in C:\projects\cake\src\Cake.Core\CakeEngine.cs:line 436
```

Ideas, @gep13, @arturcic?

**Comment by gep13** [comment]
@asbjornu what command are you running when you get this error?

**Comment by arturcic** [comment]
@gep13 `./build.ps1 -Target Build-Documentation`

**Comment by gep13** [comment]
Ah gotcha, you mean this error is coming up when running the new docs workflow? On phone, so hard to view that.

Could it be due to not everything being checked up? In the build.hmm there is an additional checkout step, which is likely required to fully assert version number.

Arguably this isn't needed for a docs build, so could be that we make that part of the Setup task conditional?

**Comment by asbjornu** [comment]
> Arguably this isn't needed for a docs build, so could be that we make that part of the Setup task conditional?

Good point, @gep13. If I only knew where `Task("Setup")` was defined… 😅

**Comment by gep13** [comment]
@asbjornu this isn't a Task, but rather a setup method:

https://github.com/gittools/gitversion/blob/main/build.cake#L55

**Comment by asbjornu** [comment]
Gotcha @gep13. Thanks!

File: docs/input/docs/usage/cli/arguments.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    /nofetch        Disables 'git fetch' during version calculation. Might cause
+                    GitVersion to not calculate your version as expected.
+
+# Execute build args
+    /exec           Executes target executable making GitVersion variables
+                    available as environmental variables
+    /execargs       GitVersionOptions for the executable specified by /exec
+    /proj           Build a msbuild file, GitVersion variables will be passed as
+                    msbuild properties
+    /projargs       Additional arguments to pass to msbuild
```
Comment (arturcic) [inline docs/input/docs/usage/cli/arguments.md:97]: These were already removed from the code

**Review by arturcic — CHANGES_REQUESTED** [review]

**Comment by arturcic** [comment]
@asbjornu look really great the new UI, really fresh. I had a quick review, there is one comment I added but overall looks good. I think we can merge it as it is now, later we can/need to improve with follow up PRs

**Review by arturcic — APPROVED** [review]

**Pull request merged**